File: DLL/CCRC/CCRC/SqlHelper.vb
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Data.SqlClient
Imports Microsoft.Win32
Imports System.IO
Public NotInheritable Class Sqlhelper


    Public Sub New()
    End Sub

    Public Function GetConnectionString() As SqlConnection
        Dim MyConnection As New SqlConnection(GetConnectionStringFromRegistry())
        Return MyConnection
    End Function

    Public Function GetReader(ByVal SQL As String) As SqlDataReader
        Dim rd As SqlDataReader = Nothing

        Dim MyConnection As SqlConnection = GetConnectionString()

        Using selectCommand As SqlCommand = MyConnection.CreateCommand()
            selectCommand.CommandText = SQL
            ' 
            selectCommand.Connection.Open()
            rd = selectCommand.ExecuteReader(CommandBehavior.CloseConnection)
        End Using


        Return rd
    End Function

    Public Sub ExecuteNonQuery(ByVal SQL As String)
        Using MyConnection As SqlConnection = GetConnectionString()
            Using command As New SqlCommand(SQL)
                command.Connection = MyConnection
                command.Connection.Open()
                command.ExecuteNonQuery()
                command.Connection.Close()
            End Using
        End Using
    End Sub

    Public Sub ExecuteNonQuery(ByVal storedProcedureName As String, ByRef outputParam As Integer, ByVal ParamArray parameterValues As Object())
        Using MyConnection As SqlConnection = GetConnectionString()
            Using command As New SqlCommand(storedProcedureName)
                command.CommandType = CommandType.StoredProcedure
                command.Connection = MyConnection
                For i As Integer = 0 To parameterValues.Length - 1
                    command.Parameters.Add(parameterValues(i))
                Next
                command.Connection.Open()
                command.ExecuteNonQuery()

                outputParam = CInt(command.Parameters(0).Value)
            End Using
        End Using
    End Sub


    Public Function ExecuteScalar(ByVal SQL As String, ByRef outputParam As String) As String
        Using MyConnection As SqlConnection = GetConnectionString()
            Using command As New SqlCommand(SQL)
                command.Connection = MyConnection
                command.Connection.Open()
                outputParam = command.ExecuteScalar().ToString()
                Return outputParam
            End Using
        End Using
    End Function
    Public Function GetConnectionStringFromRegistry() As String

        Dim myCo As RegistryKey
        Dim rootKey As String
        Dim conStr As String = ""
        rootKey = "SOFTWARE\MYCompany"
        myCo = Registry.LocalMachine.OpenSubKey(rootKey)
        If myCo IsNot Nothing Then
            conStr = DirectCast(myCo.GetValue("DOTNET_CONNECTION_STRING", ""), String)
            myCo.Close()
        End If
        Return conStr
    End Function
End Class

File: DLL/CCRCSecure/Sqlhelper.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Data.SqlClient;
using CCRCEncryption;
using System.IO;
public sealed class Sqlhelper
{
  

    public Sqlhelper()
    {
    }

    public SqlConnection GetConnectionString()
    {
        SqlConnection MyConnection = new SqlConnection(SecureConnection.GetConnectionStringFromRegistry ());
        return MyConnection;
    }

    public SqlDataReader GetReader(string SQL)
    {
        SqlDataReader rd = null;
    
            SqlConnection MyConnection = GetConnectionString();

            using (SqlCommand selectCommand = MyConnection.CreateCommand())
            {
                selectCommand.CommandText = SQL;// 
                selectCommand.Connection.Open();
                rd = selectCommand.ExecuteReader(CommandBehavior.CloseConnection);
            }


        return rd;
    }

    public void ExecuteNonQuery(string SQL)
    {
        using (SqlConnection MyConnection = GetConnectionString())
        {
            using (SqlCommand command = new SqlCommand(SQL))
            {
                command.Connection = MyConnection;
                command.Connection.Open();
                command.ExecuteNonQuery();
                command.Connection.Close();
            }
        }
    }

    public void ExecuteNonQuery(string storedProcedureName, out int outputParam, params object[] parameterValues)
    {
            using (SqlConnection MyConnection = GetConnectionString())
            {
                using (SqlCommand command = new SqlCommand(storedProcedureName))
                {
                    command.CommandType = CommandType.StoredProcedure;
                    command.Connection = MyConnection;
                    for (int i = 0; i < parameterValues.Length; i++)
                    {
                        command.Parameters.Add(parameterValues[i]);
                    }
                    command.Connection.Open();
                    command.ExecuteNonQuery();

                    outputParam = (int)command.Parameters[0].Value;
                }
            }
    }


    public string ExecuteScalar(string SQL, out string outputParam)
    {
        using (SqlConnection MyConnection = GetConnectionString())
        {
            using (SqlCommand command = new SqlCommand(SQL))
            {
                command.Connection = MyConnection;
                command.Connection.Open();
                outputParam = command.ExecuteScalar().ToString();
                return outputParam;
            }
        }
    }




}